src/charm.py

```diff
@@ -1015,6 +1016,35 @@ def _unit_ip(self) -> str:
         """Current unit ip."""
         return str(self.model.get_binding(PEER).network.bind_address)
 
+    @property
+    def _database_ip(self) -> str:
+        """Database endpoint address."""
+        return str(self.model.get_binding(DATABASE).network.bind_address)
+
+    @property
+    def listen_ips(self) -> list[str]:
+        """Return the IPs to listen on.
+
+        This is used to configure the PostgreSQL server.
+        Peer relation IP must be first in list.
+        ref.: https://patroni.readthedocs.io/en/latest/yaml_configuration.html#postgresql
+        """
+        if self._unit_ip == self._database_ip:
+            return [self._unit_ip]
```

Comment:
Original idea was to use a set and skip this test, but order is important on patroni file, from [docs](https://patroni.readthedocs.io/en/latest/yaml_configuration.html#postgresql):

> Patroni will use the first address from this list to establish local connections to the PostgreSQL node.